### PR TITLE
Sort users by name on time tracking page dropdown

### LINKED ISSUE #1884: Sort users by name on time tracking page dropdown

### PR DESCRIPTION
**Pull request by apoorv1316**
Fixes #1884

### CODE REVIEW

**Review by keshavbiswa — COMMENTED** [review]
It seems like we're duplicating the code, is it possible to extract this out to a reusable module?

**Comment by apoorv1316** [comment]
@keshavbiswa Please review

File: app/services/employee_fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module EmployeeFetcher
```
Comment (akhilgkrishnan) [inline app/services/employee_fetcher.rb:3]: Seems like this one is a helper or concern, but its not a service.

Comment (apoorv1316) [reply app/services/employee_fetcher.rb:3]: Yeah we can put it in app/services/concerns like we do for model and controllers. But then we have to autoload that. Should I do that?

Comment (apoorv1316) [reply app/services/employee_fetcher.rb]: moved it to concerns

File: app/services/concerns/employee_fetching_concern.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module EmployeeFetchingConcern
+  extend ActiveSupport::Concern
+
+  included do
+    def set_employees
+      @employees = admin? ? current_company_users : [current_user]
```
Comment (keshavbiswa) [inline app/services/concerns/employee_fetching_concern.rb:8]: There is an inconsistency of data being sent here. 

If `admin` is true:, we're sending only :id, :first_name, :last_name, but if it's false then we're sending `current_user` to the array. 

Can you check if it's possible to just send id, first_name, last_name or just send the whole object?

File: config/application.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@ class Application < Rails::Application
     # not contain `.rb` files, or that should not be reloaded or eager loaded.
     # Common ones are `templates`, `generators`, or `middleware`, for example.
     config.autoload_lib(ignore: %w(assets tasks))
+    config.autoload_paths += %W(#{config.root}/app/services/concerns)
```
[the remark this answers is not in the document]
Comment (apoorv1316) [reply config/application.rb:20]: %w does not allow string interpolation. So %W is correctly used here

File: app/services/concerns/employee_fetching_concern.rb (hunk 2 above)
Comment (apoorv1316) [reply app/services/concerns/employee_fetching_concern.rb]: done

**Review by keshavbiswa — APPROVED** [review]
Looks good, can we replace the slices of data with the full data itself? Seems easier that way and makes it reusable.

**Pull request merged**